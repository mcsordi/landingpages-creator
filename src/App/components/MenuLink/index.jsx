import P from 'prop-types';

function MenuLink({ text, link, newTab = false }) {
  return (
    <div>
      <a
        className="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
        href={link}
        target={newTab ? '_blank' : '_self'}
        rel="noreferrer"
      >
        {text}
      </a>
    </div>
  );
}
MenuLink.propTypes = {
  link: P.string.isRequired,
  newTab: P.bool,
  text: P.string.isRequired,
};
export default MenuLink;
