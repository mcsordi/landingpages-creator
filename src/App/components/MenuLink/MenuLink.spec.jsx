import { render, screen } from '@testing-library/react';
import MenuLink from '.';

describe('<MenuLink/>', () => {
  it('should render MenuLink with default properties', () => {
    render(<MenuLink link="https://www.google.com.br">text</MenuLink>);
    const component = screen.getByRole('link');
    expect(component).toHaveAttribute('target', '_self');
  });
  it('should render MenuLink with blank property', () => {
    render(
      <MenuLink newTab link="https://www.google.com.br">
        text
      </MenuLink>,
    );
    const component = screen.getByRole('link');
    expect(component).toHaveAttribute('target', '_blank');
  });
  it('should render MenuLink with underline class', () => {
    render(<MenuLink link="https://www.google.com.br">text</MenuLink>);
    const component = screen.getByRole('link');
    expect(component).toHaveClass('hvr-underline-from-center');
  });
});
