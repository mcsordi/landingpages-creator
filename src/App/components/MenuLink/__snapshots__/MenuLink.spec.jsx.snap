// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink/> should render MenuLink with underline class 1`] = `
<a
  class="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
  href="https://www.google.com.br"
  rel="noreferrer"
  target="_self"
/>
`;
